src/lib/holocene/date-picker.svelte

```diff
@@ -46,6 +46,8 @@
       const date = parseInt(inputDateSplit[1]);
       const newDate = new Date(year, month, date);
       dispatch('datechange', newDate);
+    } else if (inputDate.length === 0) {
+      dispatch('dateclear', {});
```

Comment:
I think we'll want to do this when you click the x and clear the whole input as well. That's more likely than backspacing the full date.

Reply:
The `Input` has a [clear](https://github.com/temporalio/ui/blob/main/src/lib/holocene/input/input.svelte#L83) event we could utilize here.